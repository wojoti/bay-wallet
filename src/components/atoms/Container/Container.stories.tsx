import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {Container} from 'components/atoms';
import {Text, View} from 'react-native';
import {ContainerProps} from './Container';

export default {
  title: 'Atoms/Container',
  component: Container,
  decorators: [
    Story => (
      <View style={{borderWidth: 1}}>
        <Story />
      </View>
    ),
  ],
  argTypes: {
    justify: {
      control: {type: 'select'},
      options: [
        'flex-start',
        'flex-end',
        'center',
        'space-between',
        'space-around',
        'space-evenly',
        undefined,
      ],
    },
    align: {
      control: {type: 'select'},
      options: [
        'flex-start',
        'flex-end',
        'center',
        'stretch',
        'baseline',
        undefined,
      ],
    },
    background: {
      control: {type: 'color'},
    },
  },
} as ComponentMeta<typeof Container>;
const Template: ComponentStory<typeof Container> = (args: ContainerProps) => (
  <Container {...args}>
    <Text>Example text</Text>
    <Text>Example text</Text>
    <Text>Example text</Text>
  </Container>
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  align: 'flex-start',
  justify: 'flex-start',
  flex: 0,
};
