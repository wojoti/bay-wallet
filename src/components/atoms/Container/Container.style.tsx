import {StyleSheet} from 'react-native';

interface Props {
  justify?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | undefined;
  align?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'stretch'
    | 'baseline'
    | undefined;
  flex?: number;
  direction?: 'row' | 'column';
}
const style = ({justify, align, flex, direction}: Props) =>
  StyleSheet.create({
    wrapper: {
      justifyContent: justify || 'flex-start',
      alignItems: align || 'stretch',
      flex: flex,
      flexDirection: direction || 'column',
      width: '100%',
    },
  });
export default style;
