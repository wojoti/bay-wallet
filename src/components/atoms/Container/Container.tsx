import {PropsWithChildren} from 'react';
import {View} from 'react-native';
import style from './Container.style';

export type ContainerProps = PropsWithChildren<{
  justify?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | undefined;
  align?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'stretch'
    | 'baseline'
    | undefined;
  flex?: number;
  testId?: string;
  direction?: 'row' | 'column';
  customStyle?: object;
}>;

const Container = ({
  justify,
  align,
  flex,
  direction,
  testId,
  customStyle,
  children,
}: ContainerProps) => {
  return (
    <View
      style={[style({justify, align, flex, direction}).wrapper, customStyle]}
      testID={testId}>
      {children}
    </View>
  );
};

export default Container;
