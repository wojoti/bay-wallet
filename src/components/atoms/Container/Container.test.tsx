import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import Container, {ContainerProps} from './Container';

const props: ContainerProps = {
  testId: 'test-container-id',
};

const modifiedProps: ContainerProps = {
  ...props,
  flex: 1,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<Container {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render Container', () => {
  render(<Container {...props} />);
  const ContainerElement = screen.getByTestId('test-container-id');
  expect(ContainerElement).toBeOnTheScreen();
});

test('should render container with default style', () => {
  render(<Container {...props} />);
  const containerElement = screen.getByTestId('test-container-id');
  expect(containerElement).toBeOnTheScreen();
  expect(containerElement).toHaveStyle({
    flexDirection: 'column',
  });
});

test('should render container with modified style through props', () => {
  render(<Container {...modifiedProps} />);
  const containerElement = screen.getByTestId('test-container-id');
  expect(containerElement).toBeOnTheScreen();
  expect(containerElement).toHaveStyle({flex: 1});
});
