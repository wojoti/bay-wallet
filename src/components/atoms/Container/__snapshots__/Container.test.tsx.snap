// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="test-container-id"
/>
`;
